src/pages/createConfig/step2AddConfig/step2AddConfig.tsx

```diff
@@ -18,31 +18,30 @@ type Step2AddConfigProps = {
   initialJsonStringData?: string | undefined;
 };
 export const Step2AddConfig: React.FC<Step2AddConfigProps> = ({ onDataChange, onJsonStringChange, schemaId, initialJsonStringData }) => {
-  const fetchSchemaDereference = useCallback(() => getSchema({ id: schemaId, shouldDereference: true }), [schemaId]);
+  const fetchSchemaDereference = useCallback(() => getSchema({ id: schemaId, shouldDereference: false }), [schemaId]);
   const { data: schema } = useQuery({
-    queryKey: ['getSchemaWitheRefs'],
+    queryKey: [getSchema.name, schemaId],
     queryFn: fetchSchemaDereference,
     enabled: !!schemaId,
   });
 
   const [isFetching, setIsFetching] = useState<boolean>(false);
+  const [dereferencedSchema, setDereferencedSchema] = useState<SchemaObject>({});
   const [errors, setErrors] = useState<ErrorCardProps[]>([]);
 
-  const validateJson = useMemo(() => {
-    const validate = ajvInstance.compile(schema ?? {});
-    return async (data: unknown) => {
-      const isValid = validate(data);
-      return { isValid, errorMessage: validate.errors?.[0].message };
-    };
+  useEffect(() => {
+    if (!schema) {
+      return;
+    }
+    dereferenceJsonSchema(schema).then(setDereferencedSchema);
```

Comment:
what if this fails?
you can use dereferenceJsonSchema with react-query to handle this.
https://tanstack.com/query/v4/docs/framework/react/guides/dependent-queries#usequery-dependent-query

Reply:
done